### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by tristantarrant**
This PR contains two parts:
* replaces all links to JavaDocs and WildFly docs using a parameterized prefix
* splits and rearranges the server guide and adds some more details about the JMX connector

### CODE REVIEW

File: documentation/src/main/asciidoc/user_guide/marshalling.adoc [hunk 1]
```diff
@@ -13,13 +13,13 @@ Since performance is a very important factor in this process, Infinispan uses JB
...
 
-TIP: Since Infinispan 5.0, marshalling non-Serializable key/value objects is supported as long as users can to provide meaningful Externalizer implementations for these non-Seralizable objects. <<_plugging_infinispan_with_user_defined_externalizers, This section>> has more details. 
+TIP: Since Infinispan 5.0, marshalling non-Serializable key/value objects is supported as long as users can to provide meaningful Externalizer implementations for these non-Seralizable objects. <<_plugging_infinispan_with_user_defined_externalizers, This section>> has more details.
 
 If you're unable to retrofit Serializable or Externalizable into the classes whose instances are stored in Infinispan, you could alternatively use something like link:http://x-stream.github.io/[XStream] to convert your Non-Serializable objects into a String that can be stored into Infinispan. You can find an example on how to use XStream link:http://anonsvn.jboss.org/repos/infinispan/trunk/core/src/test/java/org/infinispan/marshall/TestObjectStreamMarshaller.java[here] . The one caveat about using XStream is that it slows down the process of storing key/value objects due to the XML transformation that it needs to do.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/user_guide/marshalling.adoc]: This link is really old, and the newest `TestObjectStreamMarshaller` doesn't use XStream any more...

File: documentation/src/main/asciidoc/user_guide/marshalling.adoc [hunk 2]
```diff
@@ -45,22 +45,22 @@ builder.storeAsBinary().enable();
...
+It is this wrapper class that transparently takes care of serialization and deserialization on demand, and internally may have a reference to the object itself being wrapped, or the serialized, byte array representation of this object.
 
 This has a particular effect on the behavior of equality.
 The equals() method of this class will either compare binary representations (byte arrays) or delegate to the wrapped object instance's equals() method, depending on whether both instances being compared are in serialized or deserialized form at the time of comparison.
 If one of the instances being compared is in one form and the other in another form, then one instance is either serialized or deserialized.
-The preference will be to compare object representations, unless the cache is link:http://docs.jboss.org/infinispan/{infinispanversion}/apidocs/org/infinispan/Cache.html#compact()[compacted], in which case byte array comparison is favored.
+The preference will be to compare object representations, unless the cache is link:{javadocroot}/org/infinispan/Cache.html#compact()[compacted], in which case byte array comparison is favored.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/user_guide/marshalling.adoc]: Old-style javadoc link

File: documentation/src/main/asciidoc/user_guide/server_protocols_hotrod.adoc [hunk 3]
```diff
@@ -943,7 +943,7 @@ To switch to the initial servers defined in the client configuration, call
...
+Data structures, such as Infinispan link:{javadocroot}/org/infinispan/Cache.html[Cache] , that are accessed and modified concurrently can suffer from data consistency issues unless there're mechanisms to guarantee data correctness. Infinispan Cache, since it implements link:http://docs.oracle.com/javase/6/docs/api/java/util/concurrent/ConcurrentMap.html[ConcurrentMap] , provides operations such as link:http://docs.oracle.com/javase/6/docs/api/java/util/concurrent/ConcurrentMap.html#replace(K, V, V)[conditional replace] , link:http://docs.oracle.com/javase/6/docs/api/java/util/concurrent/ConcurrentMap.html#putIfAbsent(K, V)[putIfAbsent] , and link:http://docs.oracle.com/javase/6/docs/api/java/util/concurrent/ConcurrentMap.html#remove(java.lang.Object, java.lang.Object)[conditional remove] to its clients in order to guarantee data correctness. It even allows clients to operate against cache instances within JTA transactions, hence providing the necessary data consistency guarantees.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/user_guide/server_protocols_hotrod.adoc]: Old-style javadoc links

File: documentation/src/main/asciidoc/faqs/faqs.asciidoc [hunk 4]
```diff
@@ -16,56 +16,56 @@ any questions you may have on the link:http://www.infinispan.org/community[Infin
 === What is Infinispan?
 
 Infinispan is an open source data grid platform.
-It exposes a link:$$http://jcp.org/en/jsr/detail?id=107$$[JSR-107] compatible link:$$http://docs.jboss.org/infinispan/6.0/apidocs/org/infinispan/Cache.html$$[`Cache`] interface (which in turn extends `java.util.Map`) in which you can store objects.
+It exposes a link:{javadocroot}/org/infinispan/Cache.html$$[`Cache`] interface (which in turn extends `java.util.Map`) in which you can store objects.
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/faqs/faqs.asciidoc]: We don't actually need to use any notation unless special characters get escaped by asciidoc

File: documentation/src/main/asciidoc/faqs/faqs.asciidoc [hunk 5]
```diff
@@ -612,19 +611,19 @@ See link:$$https://docs.jboss.org/author/pages/viewpage.action?pageId=3737110$$[
...
+* Or, make sure you create a namedCache for `___hotRodTopologyCache` with sync replication, state transfer, no expiration and no eviction.
 
 ==== Is there a way to do a Bulk Get on a remote cache?
-There's no bulk get operation in Hot Rod, but the Java Hot Rod client has implemented via link:$$http://docs.jboss.org/infinispan/6.0/apidocs/org/infinispan/client/hotrod/RemoteCache.html$$[RemoteCache] the getAsync() operation, which returns a link:$$http://docs.jboss.org/infinispan/6.0/apidocs/org/infinispan/util/concurrent/NotifyingFuture.html$$[org.infinispan.util.concurrent.NotifyingFuture] (extends java.util.concurrent.Future). So, if you want to retrieve multiple keys in parallel, just call multiple times _getAsync()_ and when you need the values, just call _Future.get()_ , or attach a link:$$http://docs.jboss.org/infinispan/6.0/apidocs/org/infinispan/util/concurrent/FutureListener.html$$[ _FutureListener_ ] to the _NotifyingFuture_ to get notified when the value is ready.
+Use the link:{javadocroot}/org/infinispan/client/hotrod/RemoteCache.html#retrieveEntries(java.lang.String,+int)[`RemoteCache#retrieveEntries`] method or one of its alternatives.
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/faqs/faqs.asciidoc]: The JavaDoc issue will be fixed when we release 9.0.0.Beta1

**Comment by tristantarrant** [comment]
I have fixed all reported issues

**Comment by danberindei** [comment]
I found a few more old-style javadoc links, check out

    find . -name '*.adoc' | xargs grep "link:[^]]*)[^]]*\[" 
    find . -name '*.adoc' | xargs grep "%28"

**Comment by tristantarrant** [comment]
Dan, I've reviewed all javadoc links and I've done some more links/anchor cleanups

**Comment by danberindei** [comment]
Integrated, thanks Tristan!

**Pull request closed without merging**